src/components/Form/Form.md

```diff
@@ -325,3 +353,313 @@ const maxAge = 65
+
+;<TestFormWrapper>
+  {({ count, setCount }) => {
+    const [finalFormData, setFinalFormData] = useState({})
+    const form = (
+      <Form
+        config={{
+          formName: 'Styleguidist example form',
+          formId: '1',
+          autocompleteOff: true,
+          fields: questionGroups[count],
+          onSubmit: async (formData) => {
+            const spreaded = { ...finalFormData, ...formData }
+            if (count < questionGroups.length - 1) {
+              // Go to the next set of questions
+              setCount(count + 1)
+              setFinalFormData(spreaded)
+            } else {
+              // Submit the form -- it will actually still have formData
+              // from the first set of questions, too
+              alert(
+                'form submission successful with values:' +
+                  JSON.stringify(spreaded)
+              )
+            }
+          },
+        }}
+      >
+        {(api) => {
+          const {
+            field,
+            getFormIsValid,
+            debugEntireFormState, // REMOVE
+          } = api
+          return (
+            <div>
+              <TitleLarge.Serif.Book500>
+                Form with dynamically changing questions
+              </TitleLarge.Serif.Book500>
+              <Spacer.H16 />
+
+              <TitleSmall.Serif.Book500>Form #{count}</TitleSmall.Serif.Book500>
+
+              {count === questionGroups.length - 1 ? (
+                <>
+                  <Spacer.H16 />
+                  <TitleSmall.Serif.Book500>
+                    This is the final form!
+                  </TitleSmall.Serif.Book500>
+                </>
+              ) : null}
+
+              <Spacer.H16 />
+
+              {field(questionNames[count])}
+
+              <Spacer.H16 />
+
+              <Button.Medium.Black disabled={!getFormIsValid()} type="submit">
+                Submit
+              </Button.Medium.Black>
+
+              <Spacer.H16 />
+              <TitleSmall.Serif.Book500>Debug section</TitleSmall.Serif.Book500>
+              <div>
+                <code>
+                  {' '}
+                  fieldErrorsState (from useFormState):{' '}
+                  {JSON.stringify(debugEntireFormState().fieldErrorsState)}{' '}
+                </code>
+              </div>
+              <div>
+                <code>
+                  {' '}
+                  fieldValuesState (from useFormState):{' '}
+                  {JSON.stringify(debugEntireFormState().fieldValuesState)}{' '}
+                </code>
+              </div>
+              <div>
+                <code>
+                  {' '}
+                  finalFormData (from the example hook here):{' '}
+                  {JSON.stringify(finalFormData)}{' '}
+                </code>
+              </div>
+            </div>
+          )
+        }}
+      </Form>
+    )
+
+    // This is how the dynamic form rerenders from scratch when submitted.
```

Comment:
line 648-662 here is key to understanding the current way to make forms with dynamic questions.

Reply:
I think you pushed again and it's now 595 -- 606 lol. Yeah, I see the odd/even stuff. I'll have a look some more on train. This prototype is definitely a helpful expression of what you're after.